src/pages/Auth/RegisterStaff.tsx

```diff
@@ -0,0 +1,152 @@
+import { yupResolver } from "@hookform/resolvers/yup";
+import React, { useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import { Link, useNavigate } from "react-router-dom";
+import * as Yup from "yup";
+import { useAuth } from "../../AuthContext";
+import FormError from "./FormError";
+
+interface FormValues {
+  firstName: string;
+  lastName: string;
+  email: string;
+  password: string;
+  confirmPassword: string;
+  programs: string;
+  counselor: boolean;
+  admin: boolean;
+}
+
+const schema = Yup.object().shape({
+  firstName: Yup.string().required("First Name is required"),
+  lastName: Yup.string().required("Last Name is required"),
+  email: Yup.string().email("Invalid email address").required("Email is required"),
+  password: Yup.string().min(5, "Password must be at least 5 characters").required("Password is required"),
+  confirmPassword: Yup.string().oneOf([Yup.ref("password")], "Passwords do not match").required("Confirm password is required"),
+  programs: Yup.string().required("Program selection is required"),
+  counselor: Yup.boolean(),
+  admin: Yup.boolean()
+});
+
+// add object for firebase uid here
+
+const RegisterStaff: React.FC = () => {
+  const { registerUser, currentUser } = useAuth();
+  const [role, setRole] = useState("");
+  const [uid, setUid] = useState(""); // State variable for storing UID
```

Comment:
define type for state variable. i.e. const [role, setRole] = useState<string>("");

Reply:
What do you mean by this comment? Your example looks like what he wrote. Typescript should be able to infer state bc it's hard coded, assuming you want a string